### DISCUSSION TITLE
Sub-heading numbering with non-numbered heading

### DISCUSSION THREAD

**Discussion by LyndonAlcock** [typst/typst#2824] [General]
### Description

Basically i would like my heading numbering as follows:
```
= Past Paper (2022/2023) 
== My first question title 
=== My first sub-question
```
to yeild
# Past Paper (2022/2023)
### 1.a) My first sub-question

this is so i can compile all my past paper answers into one consistent sheet
maybe there could be a no numbering charecter in the numbering string e.g "-1.a)"

**Comment by LyndonAlcock** [comment]
this is the closest i could get to what i want to achieve:

```
#set heading(numbering: (a,..nums)=> numbering("1.a)", ..nums))
```

but this was still an issue as it creates a bracket where i do not want one:
![image](https://github.com/typst/typst/assets/86410002/e7e88e05-cf2c-4775-a70d-e56ea2ff1f4e)

**Comment by MDLC01** [comment]
The following should work:
```typ
#set heading(numbering: (..nums) => {
  let level = nums.pos().len()
  let pattern = if level == 2 {
    "1"
  } else if level == 3 {
    "1.a)"
  }
  if pattern != none {
    numbering(pattern, ..nums)
  }
})
```

**Comment by LyndonAlcock** [comment]
thank you, close enough, i belive this is the answer I am looking for:
```
#set heading(numbering: (a, ..nums) => {
  let level = nums.pos().len()
  let pattern = if level == 1 {
    "1"
  } else if level == 2 {
    "1.a)"
  }
  if pattern != none {
    numbering(pattern, ..nums)
  }
})
```

**Comment by max397574** [comment]
I think this could also be solved with this
https://github.com/typst/typst/issues/2461
